src/encoded/locust/ff_locust.py

```diff
@@ -0,0 +1,57 @@
+import os
+import json
+import random
+from requests.auth import HTTPBasicAuth
+from locust import HttpLocust, TaskSet, task, between
+
+# IMPORTANT:
+# To best make use of this you'll want to pass in some authentication
+# To do this, create the file '<env>.json'. Navigate to the portal and
+# create an access key to be used via basic auth. Set 'username' and 'password'
+# in <env>.json to your ID and secret
+# config.json is provided. Use accordingly if you want to change routes, see 
+# locust.rst in docs
+
+
+def load_auth(loc):
+    """ loads access key from given location """
+    with open(loc, 'r') as f:
+        raw = json.load(f)
+        return raw['username'], raw['password']
+
+
+class FFUserBehavior(TaskSet):
+    """
+    Locust class for FF. Tasks and behavior for unauthenticated users
+    interacting with the portal for the purposes of performance testing
+    task decorator tells locust that it is task to be run.
+    """
+    USERNAME, PASSWORD = load_auth(os.environ['LOCUST_KEY'])
+    CONFIG = json.load(open(os.environ['LOCUST_CONFIG'], 'r'))
+    ROUTES = CONFIG['routes']
+
+    def _get(self, url):
+        """ Wrapper for self.client.get that adds auth to request
+
+        Args:
+            url: endpoint to hit
+        """
+        self.client.get(url, auth=HTTPBasicAuth(self.USERNAME, self.PASSWORD))
+
+    @task(1)
+    def random_task(self):
+        """ get a random task from config """
+        url = random.choice(self.ROUTES)
+        self._get(url)
+
+
+class Locust(HttpLocust):
```

Comment:
Should this perhaps be called FFLocust? Or even just MyLocust or something that suggests it's doing something more specific than HttpLocust, not something more general, right? The naming you picked goes more general and is the kind of name I'd give if I were trying to first repair a class before then creating an application-specific use.

Reply:
Correct. Done.